language/tools/move-cli/src/main.rs

```diff
@@ -608,6 +609,52 @@ fn explain_publish_error(
                 println!("Linking API for structs/functions of module {} has changed. Need to redeploy all dependent modules.", module_id)
             }
         }
+        VMStatus::Error(CYCLIC_MODULE_DEPENDENCY) => {
+            println!(
+                "Publishing module {} introduces cyclic dependencies.",
+                module_id
+            );
+            // find all cycles with an iterative DFS
```

Comment:
Might make sense to extract this useful logic to a `print_cycle` helper function + use it below in `move doctor`? That way, you can do the cyclic dep check once in `doctor` instead of for each module.

Reply:
Yes, this will be helpful. I guess the reason I did not do so is that it kinds of breaks the mental model --- `move doctor` calls exactly the same `bytecode_verifier::*::verify_module()` checks consecutively as the VM would do, and I kind of want to preserve the property of keeping checks in `move doctor` and the VM side in-sync.

Reply:
Fair point! It's not like it's important for `move doctor` to be fast anyway... I like your way better.